src/graphnet/models/graphs/graph_definition.py

```diff
@@ -0,0 +1,269 @@
+"""Modules for defining graphs.
+
+These are self-contained graph definitions that hold all the graph-altering
+code in graphnet. These modules define what the GNNs sees as input and can be
+passed to dataloaders during training and deployment.
+"""
+
+
+from typing import Any, List, Optional, Dict, Callable
+import torch
+from torch_geometric.data import Data
+import numpy as np
+
+from graphnet.utilities.config import save_model_config
+
+from graphnet.models.detector import Detector
+from .edges import EdgeDefinition
+from .nodes import NodeDefinition
+from graphnet.models import Model
+
+
+class GraphDefinition(Model):
+    """An Abstract class to create graph definitions from."""
+
+    @save_model_config
+    def __init__(
+        self,
+        detector: Detector,
+        node_definition: NodeDefinition,
+        edge_definition: Optional[EdgeDefinition] = None,
+        node_feature_names: Optional[List[str]] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        """Construct ´GraphDefinition´. The ´detector´ holds.
+
+        ´Detector´-specific code. E.g. scaling/standardization and geometry
+        tables.
+
+        ´node_definition´ defines the nodes in the graph.
+
+        ´edge_definition´ defines the connectivity of the nodes in the graph.
+
+        Args:
+            detector: The corresponding ´Detector´ representing the data.
+            node_definition: Definition of nodes.
+            edge_definition: Definition of edges. Defaults to None.
+            node_feature_names: Names of node feature columns. Defaults to None
+            dtype: data type used for node features. e.g. ´torch.float´
+        """
+        # Base class constructor
+        super().__init__(name=__name__, class_name=self.__class__.__name__)
+
+        # Member Variables
+        self._detector = detector
+        self._edge_definiton = edge_definition
+        self._node_definition = node_definition
+        if node_feature_names is None:
+            # Assume all features in Detector is used.
+            node_feature_names = list(self._detector.feature_map().keys())  # type: ignore
+        self._node_feature_names = node_feature_names
+        if dtype is None:
```

Comment:
If I understand this code correctly you're just getting the same behavior as if you changed the default value of dtype in the constructor to torch.float instead of None. But in a slightly more more complicated way :)

Reply:
You're absolutely right. Fixed :-)